File: lib_base/src/main/java/com/czl/lib_base/binding/viewadapter/searchview/ViewAdapter.java
package com.czl.lib_base.binding.viewadapter.searchview;

import android.annotation.SuppressLint;
import android.text.TextUtils;

import androidx.databinding.BindingAdapter;

import com.czl.lib_base.binding.command.BindingCommand;
import com.czl.lib_base.widget.MaterialSearchBar;
import com.jakewharton.rxbinding2.widget.RxTextView;
import com.mancj.materialsearchbar.adapter.SuggestionsAdapter;

import java.util.concurrent.TimeUnit;

import io.reactivex.android.schedulers.AndroidSchedulers;

/**
 * @author Alwyn
 * @Date 2020/11/4
 * @Description
 */
public class ViewAdapter {

    @BindingAdapter(value = {"onNavigationCommand", "onSearchConfirmCommand", "onSearchStateCommand"}, requireAll = false)
    public static void onSearchActionCommand(MaterialSearchBar searchBar,
                                             BindingCommand<Void> command,
                                             BindingCommand<String> bindingCommand,
                                             BindingCommand<Boolean> booleanCommand) {
        searchBar.setOnSearchActionListener(new MaterialSearchBar.OnSearchActionListener() {
            @Override
            public void onSearchStateChanged(boolean enabled) {
                if (booleanCommand != null) {
                    booleanCommand.execute(enabled);
                }
            }

            @Override
            public void onSearchConfirmed(CharSequence text) {
                if (bindingCommand != null && !TextUtils.isEmpty(text)) {
                    bindingCommand.execute(text.toString());
                }
            }

            @Override
            public void onButtonClicked(int buttonCode) {
                if (MaterialSearchBar.BUTTON_NAVIGATION == buttonCode && command != null) {
                    command.execute();
                }
            }
        });

    }

    @BindingAdapter("onSearchItemClick")
    public static void onSearchItemClick(MaterialSearchBar searchBar, SuggestionsAdapter.OnItemViewClickListener listener) {
        searchBar.setSuggestionsClickListener(listener);
    }


    @SuppressLint("CheckResult")
    @BindingAdapter("onSearchTextChangeCommand")
    public static void onSearchTextChangeCommand(MaterialSearchBar searchBar, BindingCommand<String> command) {
        RxTextView.textChanges(searchBar.getSearchEditText())
                .skip(1)
                .debounce(1L, TimeUnit.SECONDS)
                .map(CharSequence::toString)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(x -> {
                    if (command != null) {
                        command.execute(x);
                    }
                });
    }

    @BindingAdapter("searchPlaceHolder")
    public static void setSearchPlaceHolder(MaterialSearchBar bar, String text) {
        bar.setPlaceHolder(text);
    }
}

File: lib_base/src/main/java/com/czl/lib_base/binding/viewadapter/searchview/ViewAdapter.kt
package com.czl.lib_base.binding.viewadapter.searchview

import androidx.databinding.BindingAdapter
import com.czl.lib_base.binding.command.BindingCommand
import com.czl.lib_base.extension.textChangeDebounce
import com.czl.lib_base.widget.MaterialSearchBar
import com.mancj.materialsearchbar.adapter.SuggestionsAdapter

/**
 * @author Alwyn
 * @Date 2020/11/4
 * @Description
 */
object ViewAdapter {
    @JvmStatic
    @BindingAdapter(
        value = ["onNavigationCommand", "onSearchConfirmCommand", "onSearchStateCommand", "onSearchIconCommand"],
        requireAll = false
    )
    fun onSearchActionCommand(
        searchBar: MaterialSearchBar,
        command: BindingCommand<Void?>?,
        bindingCommand: BindingCommand<String?>?,
        booleanCommand: BindingCommand<Boolean?>?,
        iconCommand: BindingCommand<Void?>?
    ) {
        searchBar.setOnSearchActionListener(object : MaterialSearchBar.OnSearchActionListener {
            override fun onSearchStateChanged(enabled: Boolean) {
                booleanCommand?.execute(enabled)
            }

            override fun onSearchConfirmed(text: CharSequence) {
                bindingCommand?.execute(text.toString().trim { it <= ' ' })
            }

            override fun onButtonClicked(buttonCode: Int) {
                if (MaterialSearchBar.BUTTON_NAVIGATION == buttonCode && command != null) {
                    command.execute()
                }
                if (MaterialSearchBar.BUTTON_SPEECH == buttonCode && iconCommand != null) {
                    iconCommand.execute()
                }
            }
        })
    }

    @JvmStatic
    @BindingAdapter("onSearchItemClick")
    fun onSearchItemClick(
        searchBar: MaterialSearchBar,
        listener: SuggestionsAdapter.OnItemViewClickListener?
    ) {
        searchBar.setSuggestionsClickListener(listener)
    }

    @JvmStatic
    @BindingAdapter("onSearchTextChangeCommand")
    fun onSearchTextChangeCommand(searchBar: MaterialSearchBar, command: BindingCommand<String?>?) {
        searchBar.searchEditText.textChangeDebounce {
            command?.execute(it)
        }
    }

    @JvmStatic
    @BindingAdapter("searchPlaceHolder")
    fun setSearchPlaceHolder(bar: MaterialSearchBar, text: String?) {
        bar.setPlaceHolder(text)
    }
}